patches/Workaround_glyph_position_at_eol.patch

```diff
@@ -0,0 +1,29 @@
+Index: modules/skparagraph/src/TextLine.cpp
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
+diff --git a/modules/skparagraph/src/TextLine.cpp b/modules/skparagraph/src/TextLine.cpp
+--- a/modules/skparagraph/src/TextLine.cpp	(revision 305b7c02b90b248f000b2e48e8621368a22cf887)
++++ b/modules/skparagraph/src/TextLine.cpp	(date 1664452508330)
+@@ -1258,10 +1258,19 @@
+         return { SkToS32(utf16Index) , kDownstream };
+     }
+
++    auto lineTextRange = this->textWithNewlines();
+     PositionWithAffinity result(0, Affinity::kDownstream);
+     this->iterateThroughVisualRuns(true,
+-        [this, dx, &result]
++        [this, dx, &result, lineTextRange]
+         (const Run* run, SkScalar runOffsetInLine, TextRange textRange, SkScalar* runWidthInLine) {
++            auto intersection = lineTextRange.intersection(textRange);
++            if (textRange.width() == 2 && intersection == textRange && lineTextRange.end == textRange.end) {
++                auto text = fOwner->text(textRange);
++                if (text[0] == 13 && text[1] == 10) { // CR/LF
++                    SkASSERT(result.position != 0);
```

Comment:
Should we add this assert? In my understanding - it is zero, when there is no found position, and that can happen often. On the other end - end of the line is always mean that we should stop looking in this line.

Reply:
One of the examples - we are looking for glyph position in "/r/n" line. We should return 0 in that case probably

Reply:
It can be 0 if it's an empty line, but that case is handled above in skia code. So at current state it should never fail. 
This assert is just to ensure that we notice the change in skia code after rebase/update on a new version. 
Btw, SkAssert doesnt fail in release builds.

Reply:
> but that case is handled above in skia code

Are talking about this `(SkScalarNearlyZero(this->width()) && SkScalarNearlyZero(this->spacesWidth()))`? It looks like a workaround, not as robust check if line is empty. And in the comment we see that this condition can go away.

Reply:
> This assert is just to ensure that we notice the change in skia code after rebase/update on a new version

If we catch this assert after rebase, will our new code be invalid?

Reply:
Yes, I'm talking about `(SkScalarNearlyZero(this->width()) && SkScalarNearlyZero(this->spacesWidth()))`.

If that code above would be simply gone (without adding an alternative logic for empty lines), then we'll need to add such an alternative by ourselves.

Imagine it's not the first line in the paragraph:
```
first line in the paragraph  // first not empty line 
                             // second empty line
text on a third line ....                                             
```

If we click anywhere on a 2nd line and want to set a cursor there, we shouldn't return 0 from getGlyphPositionByOffset(). `0` would be correct only for the very first empty line. So currently we do rely on that code above `(SkScalarNearlyZero(this->width()) && SkScalarNearlyZero(this->spacesWidth()))` to handle empty lines.

Reply:
> If we catch this assert after rebase, will our new code be invalid?

Not sure what you by invalid code.
If the assert fires at some point, it means we need to handle empty lines properly. Currently, we expect that skia code above handles empty lines.

Reply:
> If the assert fires at some point, it means we need to handle empty lines properly.

Yeah, this what I meant. Without handling empty lines, our code will behave incorrectly ("invalid")